src/hooks/useDrag.tsx

```diff
@@ -436,155 +457,152 @@ export const onCreate = <E,>(
-        )
-        const newDisplayHighlights = createElement(resetDisplayHighlights, true)
-
-        return updateHighlightsSection(
-          homepageState,
-          newDisplayHighlights,
-          newHighlightOptions,
-          newHighlightErrors
-        )
-      }
-
-      return updateHighlightsSection(homepageState, [true], [val], [err])
-    }
-    case "announcement": {
-      const announcementKeyExist = !_.isEmpty(
-        frontMatter.sections.find((section) =>
-          EditorHomepageFrontmatterSection.isAnnouncements(section)
-        )
-      )
-      if (!announcementKeyExist) {
-        // should not reach here, but defensively return the original state
-        return homepageState
-      }
 
-      const announcementsIndex = frontMatter.sections.findIndex((section) =>
+    return updateHighlightsSection(homepageState, [true], [val], [err])
+  }
+  if (elemType === "announcement") {
+    const announcementKeyExist = !_.isEmpty(
+      frontMatter.sections.find((section) =>
         EditorHomepageFrontmatterSection.isAnnouncements(section)
       )
-      const announcementBlockSection: AnnouncementsFrontmatterSection = frontMatter
-        .sections[announcementsIndex] as AnnouncementsFrontmatterSection
-
-      const announcements = createElementFromTop(
-        announcementBlockSection.announcements.announcement_items,
-        val as AnnouncementOption
-      )
-
-      const resetDisplaySections = _.fill(
-        Array(displayAnnouncementItems.length),
-        false
-      )
-      const newDisplayAnnouncementItems = createElementFromTop(
-        resetDisplaySections,
-        true
-      )
-
-      const newAnnouncementErrors = createElementFromTop(
-        errors.announcementItems,
-        err
-      )
+    )
+    if (!announcementKeyExist) {
+      // should not reach here, but defensively return the original state
+      return homepageState
+    }
 
-      return updateAnnouncementSection(
+    const announcementsIndex = frontMatter.sections.findIndex((section) =>
+      EditorHomepageFrontmatterSection.isAnnouncements(section)
+    )
+    const announcementBlockSection: AnnouncementsFrontmatterSection = frontMatter
+      .sections[announcementsIndex] as AnnouncementsFrontmatterSection
+
+    const announcements = createElementFromTop(
+      announcementBlockSection.announcements.announcement_items,
+      val as AnnouncementOption
+    )
+
+    const resetDisplaySections = _.fill(
+      Array(displayAnnouncementItems.length),
+      false
+    )
+    const newDisplayAnnouncementItems = createElementFromTop(
+      resetDisplaySections,
+      true
+    )
+
+    const newAnnouncementErrors = createElementFromTop(
+      errors.announcementItems,
+      err
+    )
+
+    return updateAnnouncementSection(
+      homepageState,
+      newDisplayAnnouncementItems,
+      announcements,
+      newAnnouncementErrors,
+      announcementsIndex
+    )
+  }
+  if (elemType.startsWith("textcardcard")) {
```

Comment:
we also need the same check here - `if (!isUpdateHomepageType)`

Reply:
Added type guard in https://github.com/isomerpages/isomercms-frontend/pull/1527/commits/96d0dddc2f7d06b599739083fad3887ffef41b9c